bgpd/bgp_routemap.c

```diff
@@ -572,6 +575,101 @@ struct route_map_rule_cmd route_match_ip_route_source_prefix_list_cmd = {
 	route_match_ip_route_source_prefix_list_compile,
 	route_match_ip_route_source_prefix_list_free};
 
+/* `match mac address MAC_ACCESS_LIST' */
+
+/* Match function should return 1 if match is success else return
+   zero. */
+static route_map_result_t route_match_mac_address(void *rule,
+						  struct prefix *prefix,
+						  route_map_object_t type,
+						  void *object)
+{
+	struct access_list *alist;
+
+	if (type == RMAP_BGP) {
+		alist = access_list_lookup(AFI_L2VPN, (char *)rule);
+		if (alist == NULL)
+			return RMAP_NOMATCH;
+
+		if (prefix->u.prefix_evpn.route_type != BGP_EVPN_MAC_IP_ROUTE)
+			return RMAP_NOMATCH;
+
+		return (access_list_apply(alist, &(prefix->u.prefix_evpn.mac))
```

Comment:
```
struct prefix_eth pe = {
    .family = AF_ETHERNET,
    .prefixlen = ETH_ALEN * 8,
    .eth_addr = prefix->u.prefix_evpn.mac
};
return (access_list_apply(alist, &pe) ...
```
=> and now instead of `struct ethaddr` / void pointer it's now a nice `struct prefix`